File: Fortran/plot_l96_ens.R
source("confplot_l96_ens.R")

w <- 1024
h <-  800

fname <- paste0("l96_", fil, ".dat")
e <- matrix(readBin(fname, "numeric", nt*3), nrow=nt)

pngname <- paste0("l96_", fil, ".png")
png(pngname, w, h)
title <- paste(fil, "ne=", ne, "infl=", infl, "loc=", c.loc)
plot(1:nt, e[,1], lwd=3,
     main=title, xlab="time", ylab="L2", type="l", ylim=c(0, 3),
     cex.lab=2, cex.axis=2, cex.main=3
)
abline(h=sqrt(r), lt=2)
lines(1:nt, e[,3], lwd=3, col="blue")
lines(1:nt, e[,2], lwd=3, col="red")
legend("topleft", cex=2, legend=c("mean", "stddev", "member"),
       lty=1, lwd=3, col=c("black", "blue", "red"))
dev.off()

File: Python/plot_l96_ens.py
import numpy as np
import matplotlib.pyplot as plt
from config_l96 import nt, r, ne, fil, c_loc, infl


with open(f"l96_{fil}.dat", "rb") as f:
    e1 = np.fromfile(f, count=nt)
    e2 = np.fromfile(f, count=nt)
    st = np.fromfile(f, count=nt)

plt.rcParams["font.size"] = 18

title = f"{fil} ne={ne} infl={infl} loc={c_loc}"
fig, ax = plt.subplots()
ax.plot(np.arange(nt) + 1, e1, lw=2, c="black")
ax.plot(np.arange(nt) + 1, e2, lw=2, c="red")
ax.plot(np.arange(nt) + 1, st, lw=2, c="blue")
ax.set_title(title)
ax.set_xlabel("time")
ax.set_ylabel("L2")
ax.set_ylim([0, 3])
ax.axhline(np.sqrt(r), lw=2, ls="--", c="black")
plt.show()
